### PR TITLE
Launchpad - Add Free flow `Choose a domain` task for users on free plan

### PR DESCRIPTION
**Pull request by agrullon95**
Related to #72874

## Test / Review estimation

Test: short
Review: short

## Proposed Changes

* Add Launchpad free flow `domain_upsell` task to keep users engaged with upsell opportunities.
* `Choose a domain (Upgrade plan)` checklist task will navigate user to `domains/add` page

CleanShot 2023-02-14 at 15 03 33@2x

## Testing Instructions

* Checkout this branch
* Create or use an existing Launchpad free flow site (on free plan).
* Navigate to the launchpad and verify the `Choose a domain (Upgrade plan)` task appears on the launchpad checklist.
* Click on `Choose a domain (Upgrade plan)` task and verify it navigates to `/domains/add/{site_slug}`
* Upgrade the site plan and navigate back to the launchpad screen.
* Verify the `Choose a domain (Upgrade plan` task does not appear on the launchpad checklist.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by edanzer** [comment]
My first test on this, I did not see a plans page, and just bought a domain. And when returning to launchpad, the Choose a Domain task was still showing and incomplete. The latest change adding the query param resolved that, and this is now testing as expected. 

Two questions: 
   - Do we want to show the Choose a Domain task as complete rather than hide it? UPDATE: Looks like this had already been discussed, and it was decided to hide this task upon completion. So good here. 
   - After upgrading, we end up a post-domain-upgrade page that has buttons to 'Add Email' and 'Manage Domains'. It is not obvious, and may prove challenging, for normal users to get back to Launchpad from there. For this, we could still release as is, or we could release and then plan to tweak that screen to make it easier, or we could do a separate PR to tweak that screen (ie, show link to launchpad if launchpad screen is on) and then release this.

**Comment by Addison-Stavlo** [comment]
>  UPDATE: Looks like this had already been discussed, and it was decided to hide this task upon completion.

We ideally want to show it as completed, but lets get impact out first and iterate. Hiding it based on the plan seems like a decent starting point.

> For this, we could still release as is, or we could release and then plan to tweak that screen to make it easier, or we could do a separate PR to tweak that screen (ie, show link to launchpad if launchpad screen is on) and then release this.

Agree, these are also follow up improvements that could improve the experience.

After releasing this i1, lets continue to look at these potential follow ups and improve on this as much as we can this sprint (although, behind priority to resolving the atomic issues of coming soon and stale data).

**Review by edanzer — APPROVED** [review]

**Comment by edanzer** [comment]
Based on feedback around my comments above, just went ahead and approved this. If we decide it's worth improving navigation back to launchpad after upgrading, we can address that in a follow up.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/tasks.tsx [hunk 1]
```diff
@@ -127,6 +133,7 @@ export const launchpadFlowTasks: LaunchpadFlowTaskList = {
 	free: [
 		'setup_free',
 		'design_selected',
+		'domain_upsell',
 		'first_post_published',
 		'design_edited',
```
Comment (Addison-Stavlo) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/tasks.tsx:138]: I think we should add this conditional task to show up in all launchpad flows when on a free plan/ no custom domain, not just the free flow.

Its fair to follow up with other plans, but we should at least start with free, build, and write. (also feel free to merge this for free only and open a follow up)

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx [hunk 2]
```diff
@@ -77,6 +77,14 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep, flow }: S
 
 	const currentTask = getTasksProgress( enhancedTasks );
 	const launchTask = enhancedTasks?.find( ( task ) => task.isLaunchTask === true );
+
+	// Free flow - remove domain_upsell task if user is on paid plan
+	if ( flow === FREE_FLOW && ! site?.plan?.is_free && enhancedTasks ) {
```
Comment (Addison-Stavlo) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx:82]: Similar comment below, but we should make this upsell appear for all launchpad flows where the upsell is relevant. At the very least get `free`, `write`, and `build` enabled soon.

**Pull request merged**